show nick[user@host] in kills for remote users (user part is not shown in case the user did not have an ident reply)

diff -r 2139c147d1ab ircd/m_kill.c
--- a/ircd/m_kill.c	Tue Mar 23 23:04:59 2010 +0100
+++ b/ircd/m_kill.c	Tue Mar 23 23:15:21 2010 +0100
@@ -125,10 +125,17 @@
    * Note: "victim->name" is used instead of "user" because we may
    *       have changed the target because of the nickname change.
    */
-  sendto_opmask_butone(0, snomask,
+  if (MyUser(victim))
+    sendto_opmask_butone(0, snomask,
                        "Received KILL message for %s from %s Path: %s!%s %s",
                        get_client_name(victim, SHOW_IP), get_client_name_and_opername(sptr),
                        inpath, path, msg);
+  else
+    sendto_opmask_butone(0, snomask,
+                       "Received KILL message for %s[%s@%s] from %s Path: %s!%s %s",
+                       cli_name(victim), cli_user(victim)->realusername[0] != '~' ? cli_user(victim)->realusername : "", ircd_ntoa(&cli_ip(victim)),
+                       get_client_name_and_opername(sptr),
+                       inpath, path, msg);
   log_write_kill(victim, sptr, inpath, path, msg);
 
   /*
